src/server/routes/api/document/indra.js

```diff
@@ -19,10 +21,41 @@ const BASE_MODIFICATION_TYPES = ['Sumoylation', 'Desumoylation', 'Hydroxylation'
 const TRANSCRIPTION_TRANSLATION_TYPES = ['IncreaseAmount', 'DecreaseAmount'];
 const BINDING_TYPES = ['Complex'];
 
-const getDocuments = pairs => {
-  const getForPair = pair => {
-    let agent0 = pair[0].toUpperCase();
-    let agent1 = pair[1].toUpperCase();
+const getDocuments = templates => {
+  const getForIntn = intnTemplate => {
+    const getAgent = t => {
+      let agent;
+      let xref = getXref( t );
+      if ( xref ) {
+        agent = sanitizeId(xref.id) + '@' + xref.db;
+      }
+      else {
+        agent = t.name + '@TEXT';
+      }
+
+      return agent.toUpperCase();
+    };
+
+    // for some datasources id may be in a form like "HGNC:10937"
+    // in that cases just consider the string coming after ":"
+    const sanitizeId = id => {
+      let arr = id.split(':');
+
+      if ( arr.length == 1 ) {
+        return arr[0];
+      }
+
+      return arr[1];
+    };
+
+    const getXref = t => {
+      let validDbs = ['hgnc', 'chebi'];
+      let xref = _.find( t.dbXrefs, x => _.includes( validDbs, x.db.toLowerCase() ) );
```

Comment:
In the case of chemical, `chebi` information will be in the entity template `id` and `namespace` fields. No `dbXrefs` will exist for ChEBI.